Tests/LibGC/TestInterop.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <andrew@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "TestInterop.h"
+#include "TestHeap.h"
+#include <AK/TypeCasts.h>
+#include <LibGC-Swift.h>
+#include <LibGC/ForeignCell.h>
+#include <LibGC/Heap.h>
+
+#define COLLECT heap.collect_garbage(GC::Heap::CollectionType::CollectGarbage)
+#define COLLECT_ALL heap.collect_garbage(GC::Heap::CollectionType::CollectEverything)
+
+void test_interop()
+{
+    auto& heap = test_gc_heap();
+
+    COLLECT_ALL;
+
+    auto string = GC::ForeignRef<GC::HeapString>::allocate(heap);
```

Comment:
Who declares this exactly?
I  assume this comes from `LibGC-Swift.h`, but that means the swift compiler has to be run first, which is not immediately obvious from the CMakeLists stuff?
Also not sure how this would work with headers, swift has to run first, which means things that depend on swift types have to be strictly separated from things that the entire swift code depends on (comptime & runtime) - at least that's what it sounds like to me.

Reply:
LibGC+Swift.h is generated via a separate swiftc frontend pass before compiling any translation units. It takes a couple seconds per library as well. That's what runs the ClangImporter from the swift compiler and emits all sorts of cool annotated methods to manipulate the swift metadata, witness tables etc to hide swift types behind RAII

Reply:
It's explicitly invoked in the CMake. I yoinked a couple CMake files from apple/swift-cmake-examples and put them under Meta/CMake/Swift if you want the gory details. I opened an issue on CMake's issue tracker to see if we can get Evan to pull the header gen into CMake itself somehow so we don't have to invoke the frontend manually.

Reply:
It happens at configure time?
I see, so that solves the order issue, but the dependency issue remains

Reply:
Well. It happens as a build step. In theory I suppose yes, we can't have any Swift types used in a way that requires their full definition in any public header that gets imported by Swift. Including the interop bridging header from a header file is likely a terrible idea, as it will no doubt fail to parse from a clean build directory.

We can likely play games with these foreign cell types by forward declaring them and using them as pointers-only, if I refactor the allocation code a bit.